pandas/core/internals.py

```diff
@@ -4398,6 +4398,11 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
             fill_value = upcasted_na
 
             if self.is_null and not getattr(self.block,'is_categorical',None):
```

Comment:
Instead of this, I think change line 4129 (``get_empty_dtype_na``) to have it return ``np.dtype(np.object_)`` for 'category' (this is ONLY for the empty dtype). Then remove the 'is_categorical' check above. This may break other things though, which will need attention.